tests/Tests/Unit/RequestBuilderTest.php

```diff
@@ -337,6 +337,19 @@ public function testMethodWithoutPlaceholders()
         $this->assertEquals('some-value', $query['stringValue']);
     }
 
+    public function testMethodWithRequiredQueryParametersAndDefaultValues()
+    {
+        $message = (new MockRequestBody())
+            ->setName('')
+            ->setNumber(0);
+
+        $request = $this->builder->build(self::SERVICE_NAME . '/MethodWithRequiredQueryParameters', $message);
+        $query = Psr7\parse_query($request->getUri()->getQuery());
+
+        $this->assertEquals('', $query['name']);
+        $this->assertEquals(0, $query['number']);
```

Comment:
Just a question - what happens if no calls are made to `setName` and `setNumber`? Will the query param be empty or will the query parameter be omitted? And if that is the case, how will the API function?

Reply:
> what happens if no calls are made to setName and setNumber? Will the query param be empty or will the query parameter be omitted?

If no calls are made to the message class, the test results would be the same. If a query parameter is determined to be required (by the generator implementation) it will always be added to the query string as long as an implicit value or default value are set.